### PR TITLE
Fixed duplicate results returned on Android

### LINKED ISSUE #8: Sometimes duplicate results are returned on Android
This occurs when you search for a landmark nearby, and type the name of that landmark entirely. 
The reason is that on Android two searches are performed, and if the first "localized" search returns less than 10 items, then another "remote" search is performed which also returns that result.
There should be some filtering of duplicates here.

image

### PR DESCRIPTION
**Pull request by IjzerenHein**
New PR, this replaces the previous one

closes #8

### CODE REVIEW

File: android/src/main/kotlin/com/kiwi/reversegeocode/RNReverseGeocodeManager.kt [hunk 1]
```diff
@@ -104,19 +104,29 @@ class RNReverseGeocodeManager (reactContext: ReactApplicationContext) : ReactCon
...
+      // then add other more remote possibilities
+      if ((localAddresses != null) && (localAddresses.size < MAX_RESULTS)) {
+        var remoteAddresses = geocoder.getFromLocationName(
+          searchText,
+          MAX_RESULTS
+        )
+        if (remoteAddresses != null) {
+
+          // Filter out local-search duplicates
+          remoteAddresses = remoteAddresses.filter { ra -> localAddresses.find { la -> la.getLatitude() == ra.getLatitude() && la.getLongitude() == ra.getLongitude() } == null };
```
Comment (RobinCsl) [inline android/src/main/kotlin/com/kiwi/reversegeocode/RNReverseGeocodeManager.kt]: Could you rename `ra` to `remoteAddress` and `la` to `localAddress`, please? I know it is more verbose, but I think it reads better.

**Comment by IjzerenHein** [comment]
Sure, will do

**Comment by IjzerenHein** [comment]
I've made the changes.
cheers

**Comment by IjzerenHein** [comment]
Sweet, thanks guys for picking this up quickly 👍

**Pull request merged**